domain/src/main/java/uk/gov/hmcts/ccd/definition/store/domain/validation/casetype/CaseTypeEntityFieldLabelValidator.java

```diff
@@ -0,0 +1,128 @@
+package uk.gov.hmcts.ccd.definition.store.domain.validation.casetype;
+
+import java.util.Optional;
+
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.ccd.definition.store.domain.validation.SimpleValidationError;
+import uk.gov.hmcts.ccd.definition.store.domain.validation.ValidationErrorMessageCreator;
+import uk.gov.hmcts.ccd.definition.store.domain.validation.ValidationResult;
+import uk.gov.hmcts.ccd.definition.store.repository.entity.CaseFieldEntity;
+import uk.gov.hmcts.ccd.definition.store.repository.entity.CaseTypeEntity;
+import uk.gov.hmcts.ccd.definition.store.repository.entity.FieldEntity;
+
+import static java.lang.String.format;
+import static java.lang.String.join;
+import static java.util.Optional.ofNullable;
+import static org.apache.commons.lang3.ArrayUtils.subarray;
+
+@Component
+public class CaseTypeEntityFieldLabelValidator implements CaseTypeEntityValidator {
+
+    private static final String PLACEHOLDER_PATTERN = "^[a-zA-Z0-9_.\\]\\[]+$";
+    private static final char STARTING_PLACEHOLDER = '$';
+    private static final char OPENING_PLACEHOLDER = '{';
+    private static final char CLOSING_PLACEHOLDER = '}';
+
+    @Override
+    public ValidationResult validate(CaseTypeEntity caseType) {
+
+        ValidationResult validationResult = new ValidationResult();
+        // go over each field label and check for placeholders
+        caseType.getCaseFields().stream().forEach(caseFieldEntity -> {
+            String label = ofNullable(caseFieldEntity.getLabel()).orElse("");
+            boolean isCollecting = false;
+            String placeholderToSubstitute = "";
+            for (int scanIndex = 0; scanIndex < label.length(); scanIndex++) {
+                char c = label.charAt(scanIndex);
+                if (isStartPlaceholderAndNotCollecting(label, scanIndex, isCollecting)) {
+                    isCollecting = true;
+                } else if (isCollecting) {
```

Comment:
is there a chance to break this method? lot of nested ifs

Reply:
This has been broken down now.